notebooks/ir_scattered_light_manual_corrections/Correcting_for_Scattered_Light_in_IR_Exposures_by_Manually_Subtracting_Bad_Reads.ipynb

```diff
@@ -0,0 +1,963 @@
+    "for ext in ['flt','ima']:\n",
+    "    if os.path.exists(raw_filepath.replace('raw', ext)):\n",
+    "        os.remove(raw_filepath.replace('raw', ext))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " ##run CALWF3\n",
+    "calwf3(raw_filepath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### STEP 2: Correct the IMA product"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we walk through the process of subtracting the reads affected by Earth-limb scattered light from the cumulative science extension of the IMA. While an FLT product is also created in step 1, we do not use this file due to the poor-quality ramp fit in the presence of TVB.\n",
+    "\n",
+    "Overall, the steps are as follows:\n",
+    "\n",
+    "1. Grab the science data as a 3D array and the integration time as a 1D array from the science and time extensions of the IMA file and the associated dark reference file. \n",
+    "2. Convert science data from electrons/second to electrons.\n",
+    "3. Compute the difference between reads (the cumulative difference, in this case) to isolate counts and integration time from bad reads.\n",
+    "4. For each bad read, subtract the counts (and time) from ONLY that read from the total data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ima = fits.open(raw_filepath.replace('raw', 'ima'))\n",
+    "\n",
+    "#grab the science data as a 3D array and the integration time as a 1D array\n",
+    "cube, integ_time = diff.read_wfc3(raw_filepath.replace('raw','ima'))\n",
+    "\n",
+    "\n",
+    "#do the same for the corresponding dark image. The dark reference file used for the observation can be found in the primary header\n",
+    "dark_file = ima[0].header['DARKFILE'].replace('iref$', os.getenv('iref')+'/')\n",
+    "dark_cube, dark_time = diff.read_wfc3(dark_file)\n",
+    "\n",
+    "\n",
+    "#convert from count rate to counts\n",
+    "cube_counts = np.zeros(np.shape(cube))\n",
+    "dark_counts = np.zeros(np.shape(dark_cube))\n",
+    "for img in range(cube.shape[2]):\n",
+    "    cube_counts[:,:,img] = cube[:,:,img]*integ_time[img]\n",
+    "    dark_counts[:,:,img ] = dark_cube[:,:,img]/dark_time[img]*integ_time[img]\n",
+    "    \n",
+    "    \n",
+    "#Compute the difference between reads (the cumulative difference, in this case) to isolate counts (and integration time) from bad reads\n",
+    "difference = np.diff(cube_counts, axis = 2)\n",
+    "dark_diff = np.diff(dark_counts, axis = 2)\n",
+    "dt = np.diff(integ_time)\n",
+    "\n",
+    "#For each bad read, subtract the counts (or time) from ONLY that read from the total data\n",
+    "final_dark = dark_counts[:,:,-1]*1  \n",
```

Comment:
not sure, was done by Gabe Brammer in the code i based this on. I don't think it is necessary but I included in just in case.... I'll probably get rid of it